Content.Server/MachineLinking/System/SignallerSystem.cs

```diff
@@ -0,0 +1,33 @@
+using Content.Server.MachineLinking.Components;
+using Content.Server.MachineLinking.Events;
+using Content.Shared.Interaction;
+using JetBrains.Annotations;
+using Robust.Shared.GameObjects;
+using Content.Shared.Interaction.Events;
+
+namespace Content.Server.MachineLinking.System
+{
+    [UsedImplicitly]
+    public sealed class SignallerSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SignallerComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<SignallerComponent, UseInHandEvent>(OnUseInHand);
```

Comment:
IMO this should just be merged with the existing signal button. Given it provides very nearly identical functionality. I.e, just give the signal button a use-in-hand subscription.

The signaller entity can just the same as a button, but an item instead of a wallmount.

Reply:
The inability to pick it up would have been due to a bad interaction priority. ATM that would have to be fixed by doing something like 
```cs 
SubscribeLocalEvent<SignallerComponent, InteractHandEvent>(OnUseInHand, after: new[] { typeof(SharedItemSystem) });
```
so that the signaller interaction is handled after the item pickup interaction. The weird visual snapping/teleporting of the item was due to the fact that the client has the item system and predicts that the pickup interaction will happen, while the signaller system only exists server side.

In general I'd always argue for merging very similar components into a more generalized component, rather than having two or more very nearly redundant but slightly more specialized components. But I guess in this case its only like 20-30 lines of code so I don't feel that strongly and its probably fine.

Reply:
In that case, I'd vote to just leave it. I think I'm at my limite with what I can personally do with 0 c# knowledge and having someone else holding my hand